x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/comparison/In.java

```diff
@@ -122,34 +122,18 @@ public Attribute toAttribute() {
 
     @Override
     public ScriptTemplate asScript() {
-        StringJoiner sj = new StringJoiner(" || ");
         ScriptTemplate leftScript = asScript(value);
-        List<Params> rightParams = new ArrayList<>();
-        String scriptPrefix = leftScript + "==";
-        LinkedHashSet<Object> values = list.stream().map(Expression::fold).collect(Collectors.toCollection(LinkedHashSet::new));
-        for (Object valueFromList : values) {
-            // if checked against null => false
-            if (valueFromList != null) {
-                if (valueFromList instanceof Expression) {
-                    ScriptTemplate rightScript = asScript((Expression) valueFromList);
-                    sj.add(scriptPrefix + rightScript.template());
-                    rightParams.add(rightScript.params());
-                } else {
-                    if (valueFromList instanceof String) {
-                        sj.add(scriptPrefix + '"' + valueFromList + '"');
-                    } else {
-                        sj.add(scriptPrefix + valueFromList.toString());
-                    }
-                }
-            }
-        }
-
-        ParamsBuilder paramsBuilder = paramsBuilder().script(leftScript.params());
-        for (Params p : rightParams) {
-            paramsBuilder = paramsBuilder.script(p);
-        }
-
-        return new ScriptTemplate(format(Locale.ROOT, "%s", sj.toString()), paramsBuilder.build(), dataType());
+        // remove duplicates
+        List<Object> values = new ArrayList<>(
```

Comment:
`new LinkedHashSet(Foldables.valuesOf(list, dataType())`
This has the side effect of triggering a conversion if needed (which should expose potential bugs).